compiler/generator_go.c

```diff
@@ -0,0 +1,1253 @@
+
+    g->S[0] = p->mode == m_forward ? "0" : "env.LimitBackward";
+
+    write_failure_if(g, "int32(~S0) > c || c > int32(env.Limit)", p);
+    writef(g, "~Menv.Cursor = uint(c)~N", p);
+}
+
+static void generate_delete(struct generator * g, struct node * p) {
+
+    write_comment(g, p);
+    writef(g, "~Mif !env.SliceDel() {~N"
+              "~+~Mreturn false~N~-"
+              "~M}~N", p);
+}
+
+
+static void generate_next(struct generator * g, struct node * p) {
+
+    write_comment(g, p);
+    write_check_limit(g, p);
+    write_inc_cursor(g, p);
+}
+
+static void generate_tolimit(struct generator * g, struct node * p) {
+
+    write_comment(g, p);
+    g->S[0] = p->mode == m_forward ? "env.Limit" : "env.LimitBackward";
+    writef(g, "~Menv.Cursor = ~S0~N", p);
+}
+
+static void generate_atlimit(struct generator * g, struct node * p) {
+
+    write_comment(g, p);
+    g->S[0] = p->mode == m_forward ? "env.Limit" : "env.LimitBackward";
+    g->S[1] = p->mode == m_forward ? "<" : ">";
+    write_failure_if(g, "env.Cursor ~S1 ~S0", p);
+}
+
+static void generate_leftslice(struct generator * g, struct node * p) {
+
+    write_comment(g, p);
+    g->S[0] = p->mode == m_forward ? "env.Bra" : "env.Ket";
+    writef(g, "~M~S0 = env.Cursor~N", p);
+}
+
+static void generate_rightslice(struct generator * g, struct node * p) {
+
+    write_comment(g, p);
+    g->S[0] = p->mode == m_forward ? "env.Ket" : "env.Bra";
+    writef(g, "~M~S0 = env.Cursor~N", p);
+}
+
+static void generate_assignto(struct generator * g, struct node * p) {
+
+    write_comment(g, p);
+    g->V[0] = p->name;
+    writef(g, "~M~V0 = env.assign_to(~V0)~N", p);
+}
+
+static void generate_sliceto(struct generator * g, struct node * p) {
+
+    write_comment(g, p);
+    g->V[0] = p->name;
+    writef(g, "~M~V0 = env.SliceTo()~N"
+              "~Mif ~V0 == \"\" {~N"
+              "~+~Mreturn false~N~-~M}~N", p);
+}
+
+static void generate_address(struct generator * g, struct node * p) {
+
+    symbol * b = p->literalstring;
+    if (b != 0) {
+        write_literal_string(g, b);
+    } else {
+        write_varref(g, p->name);
+    }
+}
+
+static void generate_insert(struct generator * g, struct node * p, int style) {
+
+    int keep_c = style == c_attach;
+    write_comment(g, p);
+    if (p->mode == m_backward) keep_c = !keep_c;
+    if (keep_c) w(g, "~Mvar c = env.Cursor~N");
+    w(g, "~Mbra, ket := env.Cursor, env.Cursor~N");
+    writef(g, "~Menv.Insert(bra, ket, ", p);
+    generate_address(g, p);
+    writef(g, ");~N", p);
+    if (keep_c) w(g, "~Menv.Cursor = c~N");
+}
+
+static void generate_assignfrom(struct generator * g, struct node * p) {
+
+    int keep_c = p->mode == m_forward; /* like 'attach' */
+
+    write_comment(g, p);
+    if (keep_c) writef(g, "~Mvar c = env.Cursor~N", p);
+    /* Copying limits and cursors is necessary here because the rust
+     * borrowchecker does not like taking something from someone you are about
+     * to mutate... */
```

Comment:
Yep, that is copy/paste leftover from the rust version.  I'll remove this and any similar I find.